bolt/lib/Profile/StaleProfileMatching.cpp

```diff
@@ -223,10 +249,31 @@ class StaleMatcher {
     return Hash1.InstrHash == Hash2.InstrHash;
   }
 
+  /// Returns true if a profiled block was matched with its pseudo probe.
+  bool isPseudoProbeMatch(BlendedBlockHash YamlBBHash) {
+    return MatchedWithPseudoProbes.find(YamlBBHash.combine()) !=
+           MatchedWithPseudoProbes.end();
+  }
+
+  /// Returns the number of blocks matched with opcodes.
+  size_t getNumBlocksMatchedWithOpcodes() const { return MatchedWithOpcodes; }
+
+  /// Returns the number of blocks matched with pseudo probes.
+  size_t getNumBlocksMatchedWithPseudoProbes() const {
+    return MatchedWithPseudoProbes.size();
+  }
+
 private:
   using HashBlockPairType = std::pair<BlendedBlockHash, FlowBlock *>;
   std::unordered_map<uint16_t, std::vector<HashBlockPairType>> OpHashToBlocks;
   std::unordered_map<uint64_t, std::vector<HashBlockPairType>> CallHashToBlocks;
+  std::unordered_map<uint64_t, std::vector<const MCDecodedPseudoProbe *>>
```

Comment:
usual Q, how come we're using `std::unordered` rather than an LLVM ADT?

Reply:
I wanted to keep the mappings used within StaleMatcher consistent, but since I'm initializing pseudo probe mappings and iterating on the StaleMatcher differently anyways, I'll change these to LLVM ADT.

Reply:
@dcci DenseMap or LLVM ADT isn't generally better. we have a ton of undered_map throughout the code base, and we have benchmarked some cases carefully between the two, and DenseMap sometimes uses a lot more memory, as a result many uses of undered_map were intentional. 

If memory is of a concern (probe decoding and holding probe forest generally isn't cheap), I'm not sure DenseMap will help us.

Reply:
It's not better, but there's a use case for both of them. My question is mostly about why we use one instead of the other.

Reply:
@dcci @shawbyoung this due to a limitation of ADT maps, e.g., the following code crashes:
```  
  DenseMap<uint16_t, uint64_t> TestMap;
  TestMap[uint16_t(-1)] = 123;
```
If not that, I agree that ADT data structures would be preferred